### PR TITLE
feat: add `skip_mtls_uri_regex` support for ApisixTls

### LINKED ISSUE #1910: feat: add `skip_mtls_uri_regex` support for ApisixTls
### Description

Apache APISIX has added support for route-level mTLS. I believe we can add the same capability to Ingress. However, this requires version 3.4 or above of APISIX.

ref:

- https://github.com/apache/apisix/pull/9322

- https://github.com/apache/apisix/blob/0545a17b4198fcaa2210bd6406e9871dd09883e8/t/node/client-mtls.t#L509-L592

- https://apisix.apache.org/docs/apisix/next/tutorials/client-to-apisix-mtls/

### PR DESCRIPTION
**Pull request by aynp**
Closes #1910 

### Type of change:
- [x] New feature provided

### What this PR does / why we need it:
add `skip_mtls_uri_regex` support for ApisixTls

### Pre-submission checklist:
- [ ] Did you explain what problem does this PR solve? Or what new features have been added?
- [x] Have you added corresponding test cases?
- [ ] Have you modified the corresponding document?
- [x] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix-ingress-controller#community) first**

### CODE REVIEW

File: pkg/kube/apisix/apis/config/v2/types.go [hunk 1]
```diff
@@ -773,8 +773,9 @@ type ApisixSecret struct {
 
 // ApisixMutualTlsClientConfig describes the mutual TLS CA and verify depth
 type ApisixMutualTlsClientConfig struct {
-	CASecret ApisixSecret `json:"caSecret,omitempty" yaml:"caSecret,omitempty"`
-	Depth    int          `json:"depth,omitempty" yaml:"depth,omitempty"`
+	CASecret         ApisixSecret `json:"caSecret,omitempty" yaml:"caSecret,omitempty"`
+	Depth            int          `json:"depth,omitempty" yaml:"depth,omitempty"`
+	SkipMtlsUriRegex []string     `json:"skip_mtls_uri_regex,omitempty" yaml:"skip_mtls_uri_regex, omitempty"`
```
Comment (tao12345666333) [inline pkg/kube/apisix/apis/config/v2/types.go:778]: I think `SkipMTLSUriRegex` would be better.

**Review by tao12345666333 — COMMENTED** [review]
In test cases, it's best to add an expected failure example and then access it again successfully by adding skip_mtls_uri_regex.

**Comment by aynp** [comment]
I tried adding all tests on a single route
1. Test with correct cert
2. Test without cert fail
3. Test without cert but this time `skip_mtls_uri_regex` matches

**Comment by tao12345666333** [comment]
Sorry for the delay. please fix the CI jobs

**Comment by aynp** [comment]
@tao12345666333 can you please try rerunning the tasks? I tried running them on my own fork, and they all pass fine.

**Comment by tao12345666333** [comment]
let me merge the latest code from master branch.

**Comment by tao12345666333** [comment]
Please check the faild jobs. Thanks

**Comment by Revolyssup** [comment]
@aynp The tests are still failing and the failing tests seem relevant to your change. Can you take a look?

**Comment by AlinsRan** [comment]
We need to add documentation for this feature. 

Which version of apisix does this feature depend on?

**Comment by Revolyssup** [comment]
@AlinsRan I added the docs where it is also specified that which version supports this feature.

**Comment by AlinsRan** [comment]
@Revolyssup  The document needs to be updated:
https://github.com/apache/apisix-ingress-controller/blob/master/docs/en/latest/references/apisix_tls_v2.md#spec

The configuration example can be placed here.
https://github.com/apache/apisix-ingress-controller/blob/master/docs/en/latest/concepts/apisix_tls.md

**Review by AlinsRan — APPROVED** [review]

**Review by shreemaan-abhishek — APPROVED** [review]

**Pull request merged**